dts/bindings/sensor/nordic,sensor-stub.yaml

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2022 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+description: Simple sensor simulator
```

Comment:
I have no objections to this approach going into NCS, but I would like you to be aware of the ongoing upstream work on "emulators" as well in case they are relevant to you.

The difference between emulators and stubs as I understand it is that you can write an emulator driver and a real device driver for the same devicetree compatible, and then choose at build time whether to use the emulator or device driver. This can make DTS management simpler if you have an emulator defined for some sort of "default" sensor device, if one is applicable. I don't know your use cases, though, so I cannot comment on whether this approach is better for you.

Reply:
Sensor stub here is kind of minimal implementation. I try to implement sensor simulation with the minimal amount of code and put all the waveform generation to the application. This is because we wish to get rid of float point operation in some implementations and squeeze the FLASH usage.